### PR TITLE
Added hclip command

### PR DESCRIPTION
**Pull request by saltacc**
added an hclip command, it's similar to vclip but it teleports horizontally instead of vertically

### CODE REVIEW

**Comment by zeroeightysix** [comment]
It is far easier to not restrict hclip to the 4 cardinal directions, and use the player's yaw instead.

You can get a `Vec3d` of the player's facing direction quite easily using `Vec3d.fromPolar`:
```kotlin
val direction = Vec3d.fromPolar(0f, yaw)
```
using that, your entire `when` clause becomes:
```kotlin
updatePosition(
    x + direction.x * blocks,
    y,
    z + direction.z * blocks
)
```

**Comment by saltacc** [comment]
fixed

**Comment by 5HT2** [comment]
Wouldn't it be more useful to also have a vclip command?

**Comment by zeroeightysix** [comment]
> Wouldn't it be more useful to also have a vclip command?

There is one. Consider looking before you comment. It would, however, be a good idea to merge the two into one command that does both.

**Comment by 5HT2** [comment]
Wasn't aware of that, huh

File: src/main/kotlin/me/zeroeightsix/kami/feature/command/HClipCommand.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    override fun register(dispatcher: CommandDispatcher<CommandSource>) {
+        dispatcher register rootLiteral("hclip") {
+            float("blocks") {
+                does { ctx ->
+                    mc.player?.let {
+                        val direction = Vec3d.fromPolar(0f, it.yaw)
+                        it.updatePosition(
+                            it.x + direction.x * ("blocks".from<Float, CommandSource>(ctx)),
+                            it.y,
+                            it.z + direction.z * ("blocks".from<Float, CommandSource>(ctx))
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/feature/command/HClipCommand.kt]: Extract `blocks`:
```kt
val blocks: Float = "blocks" from ctx
it.updatePosition(...)
```

**Review by zeroeightysix — CHANGES_REQUESTED** [review]

**Comment by saltacc** [comment]
I changed it to use a variable for distance. Also, I don't think vclip and hclip should be merged into a singular module. Most utility mods that have vclip and hclip have them as separate commands so I think that we should as well in order to maintain consistency.

**Review by zeroeightysix — APPROVED** [review]
Thanks!

**Pull request merged**